src/lib/ApiEndpoint.ts

```diff
@@ -61,41 +77,133 @@ export class ApiEndpoint<R extends AnyResource, S extends Partial<ApiSetup>> {
     })
   }
 
+  /**
+   * Get a single entity
+   * /api/posts/123
+   *
+   * @param id
+   * @param resourceFilter
+   */
   async get<F extends ApiQueryResourceParameters<R>>(
     id: ResourceId,
     resourceFilter: F = EMPTY_OBJECT as F,
-  ): Promise<SingleApiResult<FilteredResource<R, F>, any>> {
+  ): Promise<ApiEntityResult<FilteredResource<R, F>, any>> {
+    return this.fetchEntity(id, resourceFilter) as any
+  }
+
+  /**
+   * Get an entity collection
+   * /api/posts/
+   *
+   * @param query
+   * @param resourceFilter
+   */
+  async fetch<Q extends ApiQueryFiltersParameters<R, S>, F extends ApiQueryResourceParameters<R>>(
+    query: Q = EMPTY_OBJECT as Q,
+    resourceFilter: F = EMPTY_OBJECT as F,
+  ): Promise<ApiCollectionResult<FilteredResource<R, F>, any>> {
+    return this.fetchCollection(query as any, resourceFilter) as any
+  }
+
+  /**
+   * Fetch a 1-to-1 relationship for an entity
+   * /api/posts/123/author
+   *
+   * @param id
+   * @param relationshipFieldName
+   * @param resourceFilter
+   */
+  async getRelationshipEntity<
+    RI extends ResourceToOneRelationshipNames<R>,
+    RR extends Extract<R[RI], AnyResource>,
+    F extends ApiQueryResourceParameters<RR>
+  >(
+    id: ResourceId,
+    relationshipFieldName: RI,
+    resourceFilter: F = EMPTY_OBJECT as F,
+  ): Promise<ApiEntityResult<FilteredResource<RR, F>, any>> {
+    const relationshipField = this.Resource.fields[relationshipFieldName] as ToOneRelationshipField<
+      RR['type']
+    >
+
+    return (this.fetchEntity(
+      id,
+      resourceFilter as any,
+      `/${relationshipFieldName}`,
+      relationshipField.type,
+    ) as unknown) as Promise<ApiEntityResult<FilteredResource<RR, F>, any>>
+  }
+
+  /**
+   * Fetch a 1-to-many relationship for an entity
+   * /api/posts/123/comments
+   *
+   * @param id
+   * @param relationshipFieldName
+   * @param query
+   * @param resourceFilter
+   */
+  async getRelationshipCollection<
```

Comment:
The name suggests any kind of relationship might be fetched as a collection, I'd opt for `getToManyReleationship` (and `getToOneRelationship`) to align the name with the relationship field decorators (@toOne/ToManyRelationship).